### ISSUE TITLE
'type' is forced by bravado-core to be present in definitions

### ISSUE THREAD

**Issue by prat0318** [Yelp/bravado-core#46]
If `definitions` object does not contain a `type` field, `validate` method throws up like so:

```
5	2015-08-28 00:47:14,022	y_app.yelp.web.pyramid_app ERROR   : 'type'
Traceback (most recent call last):
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/pyramid_swagger/tween.py", line 181, in validator_tween
    swagger_handler.handle_response(response, op_or_validators_map)
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/pyramid_swagger/tween.py", line 412, in _validate
    return f(*args, **kwargs)
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/pyramid_swagger/tween.py", line 535, in swaggerize_response
    response_spec, op, PyramidSwaggerResponse(response))
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/bravado_core/response.py", line 156, in validate_response
    validate_response_body(op, response_spec, response)
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/bravado_core/response.py", line 186, in validate_response_body
    validate_schema_object(response_body_spec, response_value)
  File "/nail/live/yelp/virtualenv_run/lib/python2.7/site-packages/bravado_core/validate.py", line 21, in validate_schema_object
    obj_type = spec['type']
KeyError: 'type'
```

From the [spec](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#definitions-object) it does not look like 'type' key presence should be mandated. Instead it should be defaulted to 'object' is properties field is non-empty imo. Also, `swagger-converter` does not add `type: object` to any of the `definitions` after conversions.

**Comment by prat0318** [comment]
Also, editor.swagger.io's petstore-simple has no `type: object` for all its definitions.

**Comment by prat0318** [comment]
As per ron from swagger: ` it's not mandatory in schema definition, but it is highly recommended (I'll fix the examples). Not having a type has an unexpected meaning to most users.`

**Comment by analogue** [comment]
The fix for #35 makes it so 'type' is required. The rationale being, definitions split up across multiple files are a little more involved to go track down and check for compliance. This is no longer just a matter of checking `#/definitions` in the root spec.

**Comment by msander** [comment]
Why can't we just guess the type based on the given keywords if it is not defined?

http://json-schema.org/latest/json-schema-validation.html

e.g.
properties -> object
items -> array
else -> string

I'm using Restlet Studio for designing my API and it also doesn't generate these 'type' values.
(I already dropped them a mail, but I think there might be more tools which could make problems here)
